controllers/dashboard_controller.go

```diff
@@ -347,18 +347,13 @@ func (r *GrafanaDashboardReconciler) onDashboardCreated(ctx context.Context, gra
 		exists = false
 	}
 
-	// Update when missing or the CR is updated
-	if exists && content.Unchanged(cr, hash) {
-		log.V(1).Info("dashboard model unchanged. skipping remaining requests")
-		return nil
-	}
-
-	remoteChanged, err := r.hasRemoteChange(exists, dashboardModel, dashWithMeta)
+	matchesStateInGrafana, err := r.matchesStateInGrafana(exists, dashboardModel, dashWithMeta)
 	if err != nil {
 		return err
 	}
 
-	if !remoteChanged {
+	if matchesStateInGrafana {
```

Comment:
One minor optimization: Could we skip this check and always apply the update? If the spec is the same, it shouldn't matter, and we save one HTTP request.

The only thing we'd need to check is if it causes excessive versions to be created